src/main/java/org/folio/circulation/resources/ItemsLostRequiringActualCostResource.java

```diff
@@ -0,0 +1,175 @@
+import static org.folio.circulation.support.AsyncCoordinationUtil.allOf;
+import static org.folio.circulation.support.http.client.CqlQuery.matchAny;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collector;
+
+import org.folio.circulation.domain.Account;
+import org.folio.circulation.domain.FeeFine;
+import org.folio.circulation.domain.ItemLostRequiringActualCostsRepresentation;
+import org.folio.circulation.domain.ItemLostRequiringCostsEntry;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.infrastructure.storage.ServicePointRepository;
+import org.folio.circulation.infrastructure.storage.feesandfines.AccountRepository;
+import org.folio.circulation.infrastructure.storage.feesandfines.FeeFineActionRepository;
+import org.folio.circulation.infrastructure.storage.inventory.ItemRepository;
+import org.folio.circulation.infrastructure.storage.loans.LoanPolicyRepository;
+import org.folio.circulation.infrastructure.storage.loans.LoanRepository;
+import org.folio.circulation.infrastructure.storage.loans.LostItemPolicyRepository;
+import org.folio.circulation.infrastructure.storage.loans.OverdueFinePolicyRepository;
+import org.folio.circulation.infrastructure.storage.users.PatronGroupRepository;
+import org.folio.circulation.infrastructure.storage.users.UserRepository;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.http.server.JsonHttpResponse;
+import org.folio.circulation.support.http.server.WebContext;
+import org.folio.circulation.support.results.Result;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+public class ItemsLostRequiringActualCostResource extends Resource {
+
+  private final String rootPath;
+
+  public ItemsLostRequiringActualCostResource(String rootPath, HttpClient client) {
+    super(client);
+    this.rootPath = rootPath;
+  }
+
+  @Override
+  public void register(Router router) {
+    RouteRegistration routeRegistration = new RouteRegistration(rootPath, router);
+    routeRegistration.getMany(this::getMany);
+  }
+
+  private void getMany(RoutingContext routingContext) {
+    final WebContext context = new WebContext(routingContext);
+    final Clients clients = Clients.create(context, client);
+    final Repositories repositories = new Repositories(clients);
+    final ItemLostRequiringActualCostsRepresentation itemRepresentation =
+      new ItemLostRequiringActualCostsRepresentation();
+
+    final Collection<String> fieldValues = List.of(
+      DECLARED_LOST.getValue(),
+      AGED_TO_LOST.getValue());
+
+    repositories.getItem().findByQuery(matchAny("status.name", fieldValues))
```

Comment:
There is a big chance to get an OutOfMemoryError here. 

I think we should filter by loans first and then join rest of the entities. 

A CQL for loans might be following:
```
status.name=="Open" and (itemStatus=="Declared lost" or (itemStatus=="Aged to lost" and agedToLostDelayedBilling.lostItemHasBeenBilled==true))
```
Then:
1. Fetch accounts for the loans (you can fetch only accounts with the  `Lost item fee (actual cost)` type)
1. Fetch lost item fee policies for the loans;
1. Filter out the loans that has an account with the expected type;
1. Filter out loans that uses set cost fee (this is actually missed in the story, could you please check this with Holly?), by executing `LostItemPolicy.getSetCostFee().isChargeable()`;
1. We also have to exclude loans that have no item fee at all (it is possible when item fee is zero but there is a processing fee > 0, if both item and processing fees == 0 then loan will be closed at this moment), execute `LostItemPolicy.hasNoLostItemFee()`.
1. Using the filtered list of loans join rest of the required entities, from the story I see following:
   1. Patron (loan user)
   1. Instance (but we need to fetch item -> holdings_record in order to get the instance id)
   1. Material type for the item
   1. Permanent item location (but I assume Holly means `Effective permanent location` here - it is `itemPermanentLocationId` but if it is undefined then it is `permanentLocationId` from the holdings record).
   1. Fee/fine owner for the item: see `org.folio.circulation.services.agedtolost.ChargeLostFeesWhenAgedToLostService#fetchFeeFineOwners` as an example how to fetch fee/fine owners for items.

Reply:
I've started working on this and the changes involved resulted in me rethinking the returned json structure.
If I am getting Loans first, which already supports having an Item attached, then I can instead just return a list of Loans, each with their relevant Item.
This should be able to have all of the data, as described in bullet point 6, either directly on Loans or on each Item within each Loan.
But are you suggesting that instead of returning the full Item or Loan object I should instead just return a set of rows with exactly only the columns for what is described in bullet point 6?

Reply:
@bohdan-suprun In the case of "_Filter out loans that uses set cost fee (this is actually missed in the story, could you please check this with Holly?), by executing LostItemPolicy.getSetCostFee().isChargeable();_", I have found that items will only age to lost when isChargeable is TRUE.

This problem happens during test writing where I want to use `.withActualCost(20.00)` without `.withSetCost(10.00)`. For the tests, when I do not specify this, then the Loan is never moved into agedToLost.

see: https://github.com/folio-org/mod-circulation/blob/master/src/main/java/org/folio/circulation/domain/policy/lostitem/LostItemPolicy.java#L151